src/test/mir-opt/const_allocation.main.ConstProp.after.64bit.mir

```diff
@@ -9,61 +9,13 @@ fn main() -> () {
         StorageLive(_1);                 // scope 0 at $DIR/const_allocation.rs:8:5: 8:8
         StorageLive(_2);                 // scope 0 at $DIR/const_allocation.rs:8:5: 8:8
         _2 = const {alloc1: &&[(Option<i32>, &[&str])]}; // scope 0 at $DIR/const_allocation.rs:8:5: 8:8
-                                         // ty::Const
-                                         // + ty: &&[(std::option::Option<i32>, &[&str])]
-                                         // + val: Value(Scalar(alloc1))
                                          // mir::Constant
                                          // + span: $DIR/const_allocation.rs:8:5: 8:8
-                                         // + literal: Const { ty: &&[(std::option::Option<i32>, &[&str])], val: Value(Scalar(alloc1)) }
+                                         // + literal: Const { ty: &&[(Option<i32>, &[&str])], val: Value(Scalar(alloc1)) }
         _1 = (*_2);                      // scope 0 at $DIR/const_allocation.rs:8:5: 8:8
         StorageDead(_2);                 // scope 0 at $DIR/const_allocation.rs:8:8: 8:9
         StorageDead(_1);                 // scope 0 at $DIR/const_allocation.rs:8:8: 8:9
         nop;                             // scope 0 at $DIR/const_allocation.rs:7:11: 9:2
         return;                          // scope 0 at $DIR/const_allocation.rs:9:2: 9:2
     }
 }
-
-alloc1 (static: FOO, size: 16, align: 8) {
```

Comment:
Please check why these are gone now. We're probably just collecting IDs to print from `::Ty`

Reply:
The problem seems to be that we [use](https://github.com/rust-lang/rust/blob/bf242bb1199e25ca2274df5c4114e0c9436b74e9/compiler/rustc_middle/src/mir/pretty.rs#L684-L694) a `TypeVisitor` when collecting allocations for pretty printing. The proper fix seems to be to use a `MutVisitor` for this, do you agree?